packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -208,6 +208,19 @@
       }, this.debounceTimeMs);
     }
 
+    /**
+     * Checks whether the current queryPreviewItem query has been saved
+     * in the queriesPreview in the state.
+     *
+     * @internal
+     *
+     * @returns True if the query has been saved.
+     */
+    protected get isSavedQuery(): boolean {
+      const previewItemQuery = this.queryPreviewInfo.query;
+      return !!this.previewResults[previewItemQuery];
+    }
```

Comment:
This property is used only once in the created hook (so is a dynamic property that doesn't need to be dynamic). You can delete it and make the check in the `created` directly.